arbor/lif_cell_group.cpp

```diff
@@ -140,7 +144,11 @@ void lif_cell_group::advance_cell(time_type tfinal, time_type dt, cell_gid_type
             if (delta == 0) continue;
             n_values += delta;
             // only exact sampling: ignore lax and never look at policy
-            for (auto t: times) samples.emplace_back(t, hdl);
+            for (auto t: times) {
+                if (t > last_time_sampled_[lid]) {
```

Comment:
is this check necessary considering you create the scheduled events with the last sampled time before?